double.go

```diff
@@ -62,6 +62,30 @@ END:
 		byte(bits>>32), byte(bits>>24), byte(bits>>16), byte(bits>>8), byte(bits))
 }
 
+func encFloat32(b []byte, v float32) []byte {
+	fv := float32(int32(v))
+	if fv == v {
+		iv := int32(v)
+		switch iv {
+		case 0:
+			return encByte(b, BC_DOUBLE_ZERO)
+		case 1:
+			return encByte(b, BC_DOUBLE_ONE)
+		}
+		if iv >= -0x80 && iv < 0x80 {
+			return encByte(b, BC_DOUBLE_BYTE, byte(iv))
+		} else if iv >= -0x8000 && iv < 0x8000 {
+			return encByte(b, BC_DOUBLE_SHORT, byte(iv>>8), byte(iv))
+		}
+
+		goto END
+	}
+
+END:
+	iv := int32(v * 1000)
```

Comment:
it should check whether it's equal `0.001*iv == v`,  and the unit test is not cover it now, see https://github.com/apache/dubbo-hessian-lite/blob/aea62f0a8e82cad885cc0f7f568e9c27ef6ea70c/src/main/java/com/alibaba/com/caucho/hessian/io/Hessian2Output.java#L778

Reply:
otherwise, it should use strconv to convert it to float64 and encode to 64 bit